packages/xstate-wallet/src/workflows/application.ts

```diff
@@ -196,7 +196,7 @@ const generateConfig = (
 
     //This could handled by another workflow instead of the application workflow
     closing: {
-      entry: actions.displayUi,
+      entry: [actions.displayUi, actions.assignRequestId],
```

Comment:
Whether or not the UI is displayed should probably be the responsibility of the closing protocol (UI shouldn't be displayed when virtually-funded)